lib/msal-browser/src/crypto/BrowserCrypto.ts

```diff
@@ -30,6 +30,15 @@ const S256_HASH_ALG = "SHA-256";
 const MODULUS_LENGTH = 2048;
 // Public Exponent
 const PUBLIC_EXPONENT: Uint8Array = new Uint8Array([0x01, 0x00, 0x01]);
+// UUID hex digits
+const UUID_CHARS = "0123456789abcdef";
+// UUID max counter
+const UUID_MAX_COUNTER = 0x3ff_ffff_ffff;
+
+const uint32Buffer = new Uint32Array(10);
+let uint32Cursor = 0;
+let uuidCounter = getUuidCounter();
+let uuidTimestamp = 0;
```

Comment:
Variables should not be defined in the global scope. If you need to track things across multiple calls I'd suggest passing them in as input parameters. 

However, if this is just to guarantee uniqueness we may not need it if the likelihood of collisions is small. We don't have a requirement that I'm aware of that these be 100% unique and these values will be reset each time the page is refreshed or the lib re-imported anyway so we can't really guarantee it.

Reply:
+1

Reply:
UUID v7 relies on the sequence (suffix), which is reset every millisecond. I don't think I can implement UUID v7 without using classes or global variables. Removing the sequence and making UUID suffix completely random defies the purpose of v7 structure and breaks weak causality.

Reply:
I don't understand, maybe we can take this offline, but the sequence is reset on each page load/new tab, yes? If so, what's the difference between resetting the sequence on each page versus resetting the sequence on each request?

Reply:
Removed global vars as we decided not to fully adhere to UUID v7 spec.